contracts/Tokens/Prime/Prime.sol

```diff
@@ -0,0 +1,572 @@
+
+            interests[market][account].rewardIndex = markets[market].rewardIndex;
+
+            uint score = _calculateScore(market, account);
+            interests[market][account].score = score;
+            markets[market].sumOfMembersScore = markets[market].sumOfMembersScore + score;
+        }
+    }
+
+    /**
+     * @notice fetch the current XVS balance of user in the XVSVault
+     * @param user the account address for which markets needs to be initialized
+     * @return xvsBalance the XVS balance of user
+     */
+    function _xvsBalanceOfUser(address user) internal view returns (uint256) {
+        (uint256 xvs, , uint256 pendingWithdrawals) = IXVSVault(xvsVault).getUserInfo(
+            xvsVaultRewardToken,
+            xvsVaultPoolId,
+            user
+        );
+        return (xvs - pendingWithdrawals);
+    }
+
+    /**
+     * @notice calculate the current score of user
+     * @param market the market for which to calculate the score
+     * @param user the account for which to calculate the score
+     * @return score the score of the user
+     */
+    function _calculateScore(address market, address user) internal returns (uint256) {
+        uint256 xvsBalanceForScore = _xvsBalanceForScore(_xvsBalanceOfUser(user));
+
+        IVToken vToken = IVToken(market);
+        uint256 borrow = vToken.borrowBalanceStored(user);
+        uint256 exchangeRate = vToken.exchangeRateStored();
+        uint256 balanceOfAccount = vToken.balanceOf(user);
+        uint256 supply = (exchangeRate * balanceOfAccount) / EXP_SCALE;
+
+        return
+            Scores.calculateScore(
+                xvsBalanceForScore,
+                _capitalForScore(xvsBalanceForScore, borrow, supply, market),
+                alphaNumerator,
+                alphaDenominator
+            );
+    }
+
+    /**
+     * @notice calcukate the current XVS balance that will be used in calculation of score
+     * @param xvs the actual XVS balance of user
+     * @return xvsBalanceForScore the XVS balance to use in score
+     */
+    function _xvsBalanceForScore(uint256 xvs) internal view returns (uint256) {
+        if (xvs > MAXIMUM_XVS_CAP) {
+            return MAXIMUM_XVS_CAP;
+        } else {
+            return xvs;
+        }
+    }
+
+    /**
+     * @notice calculate the capital for calculation of score
+     * @param xvs the actual XVS balance of user
+     * @param borrow the borrow balance of user
+     * @param supply the supply balance of user
+     * @param market the market vToken address
+     * @return capital the capital to use in calculation of score
+     */
+    function _capitalForScore(
+        uint256 xvs,
+        uint256 borrow,
+        uint256 supply,
+        address market
+    ) internal view returns (uint256) {
+        uint256 borrowCap = (xvs * markets[market].borrowMultiplier) / EXP_SCALE;
+        uint256 supplyCap = (xvs * markets[market].supplyMultiplier) / EXP_SCALE;
+
+        if (supply > supplyCap) {
```

Comment:
`supply` is defined in the underlying token. `supplyCap` is defined in XVS (because `xvs` is defined in XVS). Shouldn't we convert every amount to USD before using them?

Reply:
I was thinking of something like this:

```
supplyCapUSD = toUSD(supplyCap)
tokensUSD = toUSD(supply)

if (tokensUSD > supplyCapUSD) supply = supply * supplyCapUSD/tokensUSD
```

and something similar for the borrow side